pkg/sql/logictest/testdata/logic_test/composite_types

```diff
@@ -0,0 +1,213 @@
+statement ok
+CREATE TYPE t AS (a INT, b INT)
+
+statement ok
+DROP TYPE t
+
+statement ok
+CREATE TYPE t AS (a INT, b INT)
+
+statement error pq: relation "t" does not exist
+SELECT * FROM t
+
+statement error pq: type "test.public.t" already exists
+CREATE TABLE t (x INT)
+
+statement error pq: type "test.public.t" already exists
+CREATE TYPE t AS (a INT)
+
+statement ok
+CREATE TABLE torename (x INT)
+
+statement error pq: type "test.public.t" already exists
+ALTER TABLE torename RENAME TO t
+
+query TII
+SELECT (1, 2)::t, ((1, 2)::t).a, ((1, 2)::t).b
+----
+(1,2)  1  2
+
+statement error could not identify column \"foo\"
+SELECT ((1, 2)::t).foo
+
+statement ok
+CREATE TABLE tab (a t)
+
+statement ok
+INSERT INTO tab VALUES (NULL), ((1, 2))
+
+statement ok
+INSERT INTO tab VALUES ((1, NULL))
+
+# TODO(jordan): this should work fine.
+statement error VALUES types tuple{int, unknown} and tuple{int, int} cannot be matched
+INSERT INTO tab VALUES ((1, 2)), ((1, NULL))
+
+query TII rowsort
+SELECT a, (a).a, (a).b FROM tab
+----
+NULL   NULL  NULL
+(1,2)  1     2
+(1,)   1     NULL
+
+statement error cannot drop type \"t\" because other objects .* still depend on it
+DROP TYPE t
+
+# Nested types not supported. #90491
+statement error composite types that reference user-defined types not yet supported
+CREATE TYPE t2 AS (t1 t, t2 t)
+
+# Uncomment the below when #90491 is resolved.
```

Comment:
What's the deal with this? #90491 is this here PR.

Reply:
Oops, I meant #91779. Thanks.